test/models/API1811/DocumentDetailsSpec.scala

```diff
@@ -50,24 +50,65 @@ class DocumentDetailsSpec extends SpecBase {
         Json.toJson(emptyDocumentDetails) shouldBe expectedOutput
       }
 
-      "there is a posted penalty, so that no penalty details are provided" in {
-        val model = fullDocumentDetails.copy(penaltyType = Some("LPP1"), penaltyStatus = Some("POSTED"))
-        val expectedOutput = Json.obj(
-          "chargeType" -> "VAT Return Debit Charge",
-          "periodKey" -> "13RL",
-          "taxPeriodFrom" -> "2017-04-06",
-          "taxPeriodTo" -> "2018-04-05",
-          "chargeReference" -> "XM002610011594",
-          "mainTransaction" -> "4700",
-          "subTransaction" -> "1174",
-          "originalAmount" -> 45552768.79,
-          "outstandingAmount" -> 297873.46,
-          "clearedAmount" -> 45254895.33,
-          "items" -> Json.arr(fullLineItemDetailsOutputJson),
-          "accruingInterestAmount" -> 0.23,
-          "interestRate" -> 3
-        )
-        Json.toJson(model) shouldBe expectedOutput
+      "getAccruingPenalty" should {
+
+        "only write the accruing charge when there is an accruing and a posted penalty" in {
```

Comment:
Quite confused at this, is this supposed to be a unit test for `getAccruingPenalty`? If so it needs to call the function directly. Json reads/writes is another set of tests.

Speaking of JSON writes, could you write a test for when `documentPenaltyTotals` array is present but empty? It would be good to confirm the accruing penalty amount and penalty type are not being written.